### PR TITLE
Login layout

### PR DESCRIPTION
**Pull request by moralesbang**
## Summary

- Added login layout with validations

## Screenshots

Screen Shot 2019-08-14 at 1 34 59 PM

## Trello Card

https://trello.com/c/mLyXdRVW

### CODE REVIEW

File: react/tic-tac-toe/src/app/index.js [hunk 1]
```diff
@@ -1,11 +1,24 @@
...
+
+import Login from './screens/Login';
 
 import '../scss/application.scss';
 
-function App() {
-  return <Game />;
+class App extends Component {
+  handleSubmit = values => {
+    console.log(values);
```
Comment (LucasZibell) [inline react/tic-tac-toe/src/app/index.js]: remove console.log. Leave an empty function with a TODO

File: react/tic-tac-toe/src/app/screens/Login/index.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+import { Field, reduxForm } from 'redux-form';
+import { func } from 'prop-types';
+
+import styles from './styles.module.scss';
+
+const validate = values => {
+  const errors = {};
+  if (!values.email) {
+    errors.email = 'Por favor ingresa un email';
+  } else if (!/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(values.email)) {
```
Comment (LucasZibell) [inline react/tic-tac-toe/src/app/screens/Login/index.js]: Move the validation to a file `validationUtils.js`

File: react/tic-tac-toe/src/app/screens/Login/index.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    errors.email = 'Selecciona una dirección de email válido';
+  }
+
+  if (!values.password) {
+    errors.password = 'Selecciona una contraseña';
+  } else if (values.password.length < 8) {
```
Comment (LucasZibell) [inline react/tic-tac-toe/src/app/screens/Login/index.js]: same

File: react/tic-tac-toe/src/app/index.js (hunk 1 above)
Comment (moralesbang) [reply react/tic-tac-toe/src/app/index.js]: Error from newbie, whatever good comment 😎

File: react/tic-tac-toe/src/app/screens/Login/index.js (hunk 2 above)
Comment (moralesbang) [reply react/tic-tac-toe/src/app/screens/Login/index.js]: Cool ✌️

File: react/tic-tac-toe/src/app/screens/Login/index.js [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    const { handleSubmit } = this.props;
+    return (
+      <div className={styles.login}>
+        <form className={styles.loginForm} onSubmit={handleSubmit}>
+          <Field
+            name="email"
+            type="email"
+            component={this.renderField}
+            label="Email"
+            placeholder="e.g. john.doe@email.com"
```
Comment (alejobh) [inline react/tic-tac-toe/src/app/screens/Login/index.js]: Validation at field-level is something like this: https://redux-form.com/8.2.2/examples/fieldlevelvalidation/

Check that example, its a good way to reuse the validations.
ALSO in the case that you have many validations for just one field, create this validations in the component `constants.js` specifying the field name as key in a object, and the values must be an array containing the hierachy order of the validations something like:
```
import FORM_VALIDATIONS from '@constants/formValidations.js`

export const VALIDATIONS = {
email: [FORM_VALIDATIONS.required, FORM_VALIDATIONS.email],
...
};
```

File: react/tic-tac-toe/src/app/screens/Login/styles.module.scss [hunk 5]
```diff
@@ -0,0 +1,66 @@
+.field {
+  margin-bottom: 16px;
+}
+
+.label {
+  color: #4A5568;
+  display: block;
```
Comment (alejobh) [inline react/tic-tac-toe/src/app/screens/Login/styles.module.scss]: Is this necessary?

File: react/tic-tac-toe/src/app/screens/Login/styles.module.scss [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+  height: 100vh;
+  justify-content: center;
+  width: 100vw;
+}
+
+.login-form {
+  background-color: #FFF ;
+  border-radius: 5px;
+  box-shadow: 0 4px 6px -1px rgba(0,0,0,0.1),0 2px 4px -1px rgba(0,0,0,0.06);
+  max-width: 20rem;
```
Comment (alejobh) [inline react/tic-tac-toe/src/app/screens/Login/styles.module.scss]: Why are you using px, but here you use rem?

File: react/tic-tac-toe/src/app/screens/Login/index.js [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+    errors.email = 'Selecciona una dirección de email válido';
+  }
+
+  if (!values.password) {
+    errors.password = 'Selecciona una contraseña';
+  } else if (values.password.length < 8) {
+    errors.password = 'Debe tener al menos 8 caracteres';
+  }
+
+  return errors;
```
[the remark this answers is not in the document]
Comment (moralesbang) [reply react/tic-tac-toe/src/app/screens/Login/index.js]: Yeah! that's cool, in fact I have done that before you tell me ✌️

File: react/tic-tac-toe/src/app/index.js [hunk 8]
```diff
@@ -1,11 +1,24 @@
...
+import store from '~redux/store'; // eslint-disable-line import/no-unresolved
+
+import Login from './screens/Login';
 
 import '../scss/application.scss';
 
-function App() {
-  return <Game />;
+class App extends Component {
+  handleSubmit = values => {
```
[the remark this answers is not in the document]
Comment (moralesbang) [reply react/tic-tac-toe/src/app/index.js]: Yeah! You're right. That was a personal error caused ban understanding of `redux-form` theory.

File: react/tic-tac-toe/src/app/screens/Login/styles.module.scss (hunk 5 above)
Comment (moralesbang) [reply react/tic-tac-toe/src/app/screens/Login/styles.module.scss]: It's not. I think it's better remove it.

File: react/tic-tac-toe/src/app/screens/Login/styles.module.scss (hunk 6 above)
Comment (moralesbang) [reply react/tic-tac-toe/src/app/screens/Login/styles.module.scss]: Sorry, my fault. I have omitted this convention.

File: react/tic-tac-toe/src/utils/formValidations.js [hunk 9]
```diff
@@ -0,0 +1,11 @@
+export default {
+  required: (value) => value || (value && value.constructor.name) === 'Number'
+    ? undefined
+    : 'Required',
```
Comment (LucasZibell) [inline react/tic-tac-toe/src/utils/formValidations.js]: ` required: (value) => value ? undefined : 'Required',`

Comment (moralesbang) [reply react/tic-tac-toe/src/utils/formValidations.js]: I'm asking me what da... I was doing. Thanks @LucasZibell

**Pull request merged**